experimental/SetPartitions/src/LinearSetPartition.jl

```diff
@@ -0,0 +1,241 @@
+
+Initialize LinearSetPartition object, while simplifying the term.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d") 
+(Univariate polynomial ring in x over QQ, d)
+julia> linear_partition(Dict(set_partition([1, 2], [1, 1]) => S(4), set_partition([1, 1], [1, 1]) => 4*d))
+LinearSetPartition(Dict(set_partition([1, 2], [1, 1]) => S(4), set_partition([1, 1], [1, 1]) => 4*d))
+"""
+function linear_partition(coeffs::Dict{S, T}) where {S <: AbstractPartition, T <: RingElement}
+    return LinearSetPartition{S, T}(coeffs)
+end
+
+"""
+    coefficients(p::LinearSetPartition{S, T}) where { S <: AbstractPartition, T <: RingElement }
+
+Get the constructor field `coefficients`, the partition term, in form of a `Dict` from
+`SetPartition` to coefficient.
+""" 
+function coefficients(p::LinearSetPartition{S, T}) where { S <: AbstractPartition, T <: RingElement }
+    return p.coefficients
+end
+
+function hash(p::LinearSetPartition, h::UInt)
+    return hash(coefficients(p), h) 
+end
+
+function ==(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}) where 
+        { S <: AbstractPartition, T <: RingElement }
+        return coefficients(p) == coefficients(q)
+end
+
+function deepcopy_internal(p::LinearSetPartition, stackdict::IdDict)
+    if haskey(stackdict, p)
+        return stackdict[p]
+    end
+    q = linear_partition(deepcopy_internal(coefficients(p), stackdict))
+    stackdict[p] = q
+    return q
+end
+
+"""
+    linear_partition(term::Vector{Tuple{S, T}}) where { S <: AbstractPartition, T <: RingElement }
+
+Return a `LinearSetPartition` generated from the Vector `term` of 2-tuples,
+where the first element in the tuple is a `RingElement` and the second
+a `SetPartition`. Furthermore simplify the term before initializing
+the `LinearSetPartition` object with the corresponding dict.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d")
+(Univariate polynomial ring in x over QQ, d)
+julia> linear_partition([(set_partition([1, 1], [1, 1]), S(4)), (set_partition([1, 1], [1, 1]), 4*d)])
+LinearSetPartition(Dict(set_partition([1, 1], [1, 1]) => 4 + 4*d))
+```
+"""
+function linear_partition(term::Vector{Tuple{S, T}}) where { S <: AbstractPartition, T <: RingElement }
+    return linear_partition(Dict{S, T}(x[1] => x[2] for x in simplify_operation(term)))
+end
+
+"""
+    add(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}) where 
+        { S <: AbstractPartition, T <: RingElement }
+
+Return the addition between `p` and `q`.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d")
+(Univariate polynomial ring in x over QQ, d)
+julia> a = linear_partition([(set_partition([1, 2], [1, 1]), S(4)), 
+(set_partition([1, 1], [1, 1]), 4*d)])
+julia> add(a, a)
+LinearSetPartition(Dict(set_partition([1, 2], [1, 1]) => S(8), 
+SetPartition([1, 1], [1, 1]) => 8*d))
+```
+"""
+function add(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}) where 
```

Comment:
Why do you have both `add` and `+`? I think only having `+` would be enough here. In particular, `add` should not get exported IMO.
(same applies to `substract` and `scale`)